pkg/config/types.go

```diff
@@ -0,0 +1,97 @@
+package config
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// copied from https://github.com/kubernetes/apiserver/blob/v0.23.1/pkg/apis/config/types.go
+// because upstream struct do not have proper yaml definition for marshaling,
+// it was impossible to get the output into the right format
+
```

Comment:
Fighted with the marshal/unarmshal for so long, tried my own go templates but this was the easiest and most clean thing I could come up with